pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,357 @@
+			repoConfig.VulnContextualAnalysis = data["vuln_contextual_analysis"].(bool)
+			repoConfig.RetentionInDays = data["retention_in_days"].(int)
+		}
+
+		return repoConfig
+	}
+
+	var unpackRepositoryConfig = func(s *schema.ResourceData) RepositoryConfiguration {
+		d := &util.ResourceData{ResourceData: s}
+
+		repositoryConfig := RepositoryConfiguration{
+			RepoName: d.GetString("repo_name", false),
+		}
+
+		if _, ok := s.GetOk("repo_config"); ok {
+			repositoryConfig.RepoConfig = unpackRepoConfig(s.Get("repo_config").(*schema.Set))
+		}
+
+		if _, ok := s.GetOk("repo_paths_config"); ok {
+			repositoryConfig.RepoPathsConfig = unpackRepoPathConfig(s.Get("repo_paths_config").(*schema.Set))
+		}
+		return repositoryConfig
+	}
+
+	var packGeneralRepoConfig = func(repoConfig *GeneralRepoConfig) []interface{} {
+		if repoConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{
+			"vuln_contextual_analysis": repoConfig.VulnContextualAnalysis,
+			"retention_in_days":        repoConfig.RetentionInDays,
+		}
+
+		return []interface{}{m}
+	}
+
+	var packAllOtherArtifacts = func(otherArtifacts AllOtherArtifacts) []interface{} {
+		m := map[string]interface{}{
+			"index_new_artifacts": otherArtifacts.IndexNewArtifacts,
+			"retention_in_days":   otherArtifacts.RetentionInDays,
+		}
+
+		return []interface{}{m}
+	}
+
+	var packPatterns = func(patterns []Pattern) []interface{} {
+		var ps []interface{}
+
+		for _, pattern := range patterns {
+			p := map[string]interface{}{
+				"include":             pattern.Include,
+				"exclude":             pattern.Exclude,
+				"index_new_artifacts": pattern.IndexNewArtifacts,
+				"retention_in_days":   pattern.RetentionInDays,
+			}
+
+			ps = append(ps, p)
+		}
+
+		return ps
+	}
+
+	var packRepoPathsConfigList = func(repoPathsConfig *RepoPathsConfiguration) []interface{} {
+		if repoPathsConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{
+			"pattern":             packPatterns(repoPathsConfig.Patterns),
+			"all_other_artifacts": packAllOtherArtifacts(repoPathsConfig.OtherArtifacts),
+		}
+
+		return []interface{}{m}
+	}
+
+	var packRepositoryConfig = func(ctx context.Context, repositoryConfig RepositoryConfiguration, d *schema.ResourceData) diag.Diagnostics {
+		if err := d.Set("repo_name", repositoryConfig.RepoName); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_config", packGeneralRepoConfig(repositoryConfig.RepoConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_paths_config", packRepoPathsConfigList(repositoryConfig.RepoPathsConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+
+	var resourceXrayRepositoryConfigRead = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := RepositoryConfiguration{}
+
+		resp, err := m.(*resty.Client).R().
+			SetResult(&repositoryConfig).
+			SetPathParams(map[string]string{
```

Comment:
I forgot to mention last time. For a single param, you can use `SetPathParam("repo_name", d.Id())` which will make this more concise.

https://pkg.go.dev/github.com/go-resty/resty/v2#Request.SetPathParam

Reply:
Replaced.